tomoj/protocols/protocol_xCorrPrealignment.py

```diff
@@ -114,34 +211,61 @@ def computeXcorrStep(self, tsObjId):
 
         paramsXcorr = {
             'input': os.path.join(tmpPrefix, '%s.st' % tsId),
-            'output': os.path.join(extraPrefix, '%s.prexf' % tsId),
+            # 'output': os.path.join(extraPrefix, '%s.prexf' % tsId),
             'tiltfile': os.path.join(tmpPrefix, '%s.rawtlt' % tsId),
-            'RotationAngle': self.rotationAngle.get(),
-            'FilterSigma1': 0.03,
-            'FilterSigma2': 0.05,
-            'FilterRadius2': 0.25
+            'downsampling': self.downsampling.get(),
+            'variancefilter': self.variancefilter.get(),
+            'multiscale': self.multiscale.get(),
         }
-        argsXcorr = "-input %(input)s " \
-                    "-output %(output)s " \
-                    "-tiltfile %(tiltfile)s " \
-                    "-RotationAngle %(RotationAngle)f " \
-                    "-FilterSigma1 %(FilterSigma1)f " \
-                    "-FilterSigma2 %(FilterSigma2)f " \
-                    "-FilterRadius2 %(FilterRadius2)f"
-        self.runJob('tiltxcorr', argsXcorr % paramsXcorr)
-
-        paramsXftoxg = {
-            'input': os.path.join(extraPrefix, '%s.prexf' % tsId),
-            'goutput': os.path.join(extraPrefix, '%s.prexg' % tsId),
-        }
-        argsXftoxg = "-input %(input)s " \
-                     "-goutput %(goutput)s"
-        self.runJob('xftoxg', argsXftoxg % paramsXftoxg)
+
+        argsXcorr = "-loadangles %(tiltfile)s " \
+                    "-xcorr " \
+                    "downsampling %(downsampling)d " \
+                    "variancefilter %(variancefilter)d " \
+                    "multiscale %(multiscale)d "
+
+        if self.integerTranslation:
+            argsXcorr += 'integertranslation '
+        if self.cumulativereference:
+            argsXcorr += 'cumulativereference '
+        if self.loop:
+            argsXcorr += 'loop '
+        if self.roi:
+            argsXcorr += 'roi %d %d ' % (self.rwidth.get(), self.rheight.get())
+        if self.bandpass:
```

Comment:
Params from the protocol form must be accessed with .get() method. In this case if self.bandpass: -> if self.bandpass.get():

Reply:
I thought this was the way with boolean parameters (__BooleanParam__), I saw this in Scipion doc (https://scipion-em.github.io/docs/docs/developer/creating-a-protocol#executing-programs)